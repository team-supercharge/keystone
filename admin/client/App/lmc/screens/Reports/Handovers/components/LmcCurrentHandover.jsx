import React, { Component } from 'react'
import PropTypes from 'prop-types'
import LmcHandoverResidentItem from './LmcHandoverResidentItem.jsx'
import LmcHandoverNotes from './LmcHandoverNotes.jsx'
import AnimateHeight from 'react-animate-height'
import { BlankState, GlyphButton } from '../../../../../elemental'
import { isBrowser, isTablet } from 'react-device-detect'

export default class LmcCurrentHandover extends Component {
    state = {
        isShowingContent: true
    }

    toggleContent = () => {
        this.setState(prevState => ({
            isShowingContent: !prevState.isShowingContent
        }))
    }

    renderData () {
        const { logsByResident, notes } = this.props
        const isDesktopStyles = (isBrowser || isTablet)
        const isEmpty = (!logsByResident.length && !notes.length)

        return (
            <AnimateHeight
                duration={ 500 }
                height={ this.state.isShowingContent ? 'auto' : 0 }
            >
                { isEmpty ? (
                    <div style={styles.dataContainer}>
                        <BlankState
                            heading='Nothing to see here!'
                            style={{ width: '100%' }}
                        />
                    </div>
                ) : (
                    <div 
                        style={{ 
                            ...styles.dataContainer,
                            flexDirection: isDesktopStyles ? 'row' : 'column'
                        }}
                    >
                        <div 
                            style={isDesktopStyles ? styles.leftContainer : null}
                        >
                            { logsByResident.map((logGroup, i) => {
                                return (
                                    <div key={i}>
                                        <LmcHandoverResidentItem
                                            data={logGroup}
                                        />
                                    </div>
                                )
                            })}
                        </div>
                        <div 
                            style={isDesktopStyles ? styles.rightContainer : null}
                        >
                            <LmcHandoverNotes
                                notes={notes}
                            />
                        </div>
                    </div>
                ) }
            </AnimateHeight>
        )
    }

    render () {
        return (
            <div>
                <div style={styles.headingContainer}>
                    <h2 style={styles.heading}>
                        Current Handover
                    </h2>
                    <GlyphButton
                        className='lmc-collapse-button'
                        glyph={this.state.isShowingContent ? 'chevron-up' : 'chevron-down'}
                        onClick={this.toggleContent} 
                    />
                </div>
                <div className='lmc-theme-gradient' style={styles.divider} />
                { this.renderData() }
            </div>
        )
    }
}

const styles = {
    dataContainer: {
        display: 'flex',
        width: '100%'
    },
    divider: {
        height: 2,
        marginBottom: 22,
        width: '100%',
    },
    leftContainer: {
        width: '60%',
        paddingRight: 20
    },
    rightContainer: {
        width: '40%',
        flex: '1'
    },
    heading: {
        marginBottom: '0.3em',
        fontWeight: 300,
        textOverflow: 'ellipsis',
        hyphens: 'auto',
    },
    headingContainer: {
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'space-between'
    }
}

LmcCurrentHandover.propTypes = {
    logsByResident: PropTypes.array.isRequired,
    notes: PropTypes.array.isRequired
}