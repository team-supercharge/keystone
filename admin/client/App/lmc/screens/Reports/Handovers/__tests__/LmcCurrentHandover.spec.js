import React from 'react'
import { shallow } from 'enzyme'
import LmcCurrentHandover from '../components/LmcCurrentHandover.jsx'

describe('LmcCurrentHandover', () => {
    let wrapper
    let logs
    let notes

    beforeEach(() => {
        notes = [
            { id: 'TestId', note: 'This is a note', createdBy: 'TestCarer' }
        ]
        logs = [
            { id: 'TestId', description: 'This is a log', createdBy: 'TestCarer' }
        ]
        wrapper = shallow(
            <LmcCurrentHandover
                logs={logs}
                notes={notes}
            />
        )
    })

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })
})