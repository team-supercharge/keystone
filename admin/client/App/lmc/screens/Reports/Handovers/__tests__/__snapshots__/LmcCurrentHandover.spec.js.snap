// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LmcCurrentHandover renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <h2
        style={
          Object {
            "fontWeight": 300,
            "hyphens": "auto",
            "marginBottom": "0.3em",
            "textOverflow": "ellipsis",
          }
        }
      >
        Current Handover
      </h2>
      <GlyphButton
        className="lmc-collapse-button"
        glyph="chevron-up"
        glyphStyle={Object {}}
        onClick={[Function]}
        position="default"
      />
    </div>
    <div
      className="lmc-theme-gradient"
      style={
        Object {
          "height": 2,
          "marginBottom": 22,
          "width": "100%",
        }
      }
    />
    <AnimateHeight
      animateOpacity={false}
      animationStateClasses={
        Object {
          "animating": "rah-animating",
          "animatingDown": "rah-animating--down",
          "animatingToHeightAuto": "rah-animating--to-height-auto",
          "animatingToHeightSpecific": "rah-animating--to-height-specific",
          "animatingToHeightZero": "rah-animating--to-height-zero",
          "animatingUp": "rah-animating--up",
          "static": "rah-static",
          "staticHeightAuto": "rah-static--height-auto",
          "staticHeightSpecific": "rah-static--height-specific",
          "staticHeightZero": "rah-static--height-zero",
        }
      }
      applyInlineTransitions={true}
      delay={0}
      duration={500}
      easing="ease"
      height="auto"
      style={Object {}}
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "paddingRight": 20,
              "width": "60%",
            }
          }
        >
          <div>
            <LmcHandoverResidentItem
              data={
                Object {
                  "logs": Array [
                    Object {
                      "createdBy": "TestCarer",
                      "description": "This is a log",
                      "id": "TestId",
                    },
                  ],
                  "resident": Object {
                    "id": "TestId2",
                    "name": "TestResident",
                    "picture": "TestPicture",
                  },
                }
              }
            />
          </div>
        </div>
        <div
          style={
            Object {
              "flex": "1",
              "width": "40%",
            }
          }
        >
          <LmcHandoverNotes
            notes={
              Array [
                Object {
                  "createdBy": "TestCarer",
                  "id": "TestId",
                  "note": "This is a note",
                },
              ]
            }
          />
        </div>
      </div>
    </AnimateHeight>
  </div>,
  "nodes": Array [
    <div>
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <h2
          style={
            Object {
              "fontWeight": 300,
              "hyphens": "auto",
              "marginBottom": "0.3em",
              "textOverflow": "ellipsis",
            }
          }
        >
          Current Handover
        </h2>
        <GlyphButton
          className="lmc-collapse-button"
          glyph="chevron-up"
          glyphStyle={Object {}}
          onClick={[Function]}
          position="default"
        />
      </div>
      <div
        className="lmc-theme-gradient"
        style={
          Object {
            "height": 2,
            "marginBottom": 22,
            "width": "100%",
          }
        }
      />
      <AnimateHeight
        animateOpacity={false}
        animationStateClasses={
          Object {
            "animating": "rah-animating",
            "animatingDown": "rah-animating--down",
            "animatingToHeightAuto": "rah-animating--to-height-auto",
            "animatingToHeightSpecific": "rah-animating--to-height-specific",
            "animatingToHeightZero": "rah-animating--to-height-zero",
            "animatingUp": "rah-animating--up",
            "static": "rah-static",
            "staticHeightAuto": "rah-static--height-auto",
            "staticHeightSpecific": "rah-static--height-specific",
            "staticHeightZero": "rah-static--height-zero",
          }
        }
        applyInlineTransitions={true}
        delay={0}
        duration={500}
        easing="ease"
        height="auto"
        style={Object {}}
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "width": "100%",
            }
          }
        >
          <div
            style={
              Object {
                "paddingRight": 20,
                "width": "60%",
              }
            }
          >
            <div>
              <LmcHandoverResidentItem
                data={
                  Object {
                    "logs": Array [
                      Object {
                        "createdBy": "TestCarer",
                        "description": "This is a log",
                        "id": "TestId",
                      },
                    ],
                    "resident": Object {
                      "id": "TestId2",
                      "name": "TestResident",
                      "picture": "TestPicture",
                    },
                  }
                }
              />
            </div>
          </div>
          <div
            style={
              Object {
                "flex": "1",
                "width": "40%",
              }
            }
          >
            <LmcHandoverNotes
              notes={
                Array [
                  Object {
                    "createdBy": "TestCarer",
                    "id": "TestId",
                    "note": "This is a note",
                  },
                ]
              }
            />
          </div>
        </div>
      </AnimateHeight>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <LmcCurrentHandover
        logsByResident={
          Array [
            Object {
              "logs": Array [
                Object {
                  "createdBy": "TestCarer",
                  "description": "This is a log",
                  "id": "TestId",
                },
              ],
              "resident": Object {
                "id": "TestId2",
                "name": "TestResident",
                "picture": "TestPicture",
              },
            },
          ]
        }
        notes={
          Array [
            Object {
              "createdBy": "TestCarer",
              "id": "TestId",
              "note": "This is a note",
            },
          ]
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LmcCurrentHandover {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "logsByResident": Array [
            Object {
              "logs": Array [
                Object {
                  "createdBy": "TestCarer",
                  "description": "This is a log",
                  "id": "TestId",
                },
              ],
              "resident": Object {
                "id": "TestId2",
                "name": "TestResident",
                "picture": "TestPicture",
              },
            },
          ],
          "notes": Array [
            Object {
              "createdBy": "TestCarer",
              "id": "TestId",
              "note": "This is a note",
            },
          ],
        },
        "refs": Object {},
        "state": Object {
          "isShowingContent": true,
        },
        "toggleContent": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <h2
              style={
                Object {
                  "fontWeight": 300,
                  "hyphens": "auto",
                  "marginBottom": "0.3em",
                  "textOverflow": "ellipsis",
                }
              }
            >
              Current Handover
            </h2>
            <GlyphButton
              className="lmc-collapse-button"
              glyph="chevron-up"
              glyphStyle={Object {}}
              onClick={[Function]}
              position="default"
            />
          </div>
          <div
            className="lmc-theme-gradient"
            style={
              Object {
                "height": 2,
                "marginBottom": 22,
                "width": "100%",
              }
            }
          />
          <AnimateHeight
            animateOpacity={false}
            animationStateClasses={
              Object {
                "animating": "rah-animating",
                "animatingDown": "rah-animating--down",
                "animatingToHeightAuto": "rah-animating--to-height-auto",
                "animatingToHeightSpecific": "rah-animating--to-height-specific",
                "animatingToHeightZero": "rah-animating--to-height-zero",
                "animatingUp": "rah-animating--up",
                "static": "rah-static",
                "staticHeightAuto": "rah-static--height-auto",
                "staticHeightSpecific": "rah-static--height-specific",
                "staticHeightZero": "rah-static--height-zero",
              }
            }
            applyInlineTransitions={true}
            delay={0}
            duration={500}
            easing="ease"
            height="auto"
            style={Object {}}
          >
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "paddingRight": 20,
                    "width": "60%",
                  }
                }
              >
                <div>
                  <LmcHandoverResidentItem
                    data={
                      Object {
                        "logs": Array [
                          Object {
                            "createdBy": "TestCarer",
                            "description": "This is a log",
                            "id": "TestId",
                          },
                        ],
                        "resident": Object {
                          "id": "TestId2",
                          "name": "TestResident",
                          "picture": "TestPicture",
                        },
                      }
                    }
                  />
                </div>
              </div>
              <div
                style={
                  Object {
                    "flex": "1",
                    "width": "40%",
                  }
                }
              >
                <LmcHandoverNotes
                  notes={
                    Array [
                      Object {
                        "createdBy": "TestCarer",
                        "id": "TestId",
                        "note": "This is a note",
                      },
                    ]
                  }
                />
              </div>
            </div>
          </AnimateHeight>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <h2
              style={
                Object {
                  "fontWeight": 300,
                  "hyphens": "auto",
                  "marginBottom": "0.3em",
                  "textOverflow": "ellipsis",
                }
              }
            >
              Current Handover
            </h2>
            <GlyphButton
              className="lmc-collapse-button"
              glyph="chevron-up"
              glyphStyle={Object {}}
              onClick={[Function]}
              position="default"
            />
          </div>
          <div
            className="lmc-theme-gradient"
            style={
              Object {
                "height": 2,
                "marginBottom": 22,
                "width": "100%",
              }
            }
          />
          <AnimateHeight
            animateOpacity={false}
            animationStateClasses={
              Object {
                "animating": "rah-animating",
                "animatingDown": "rah-animating--down",
                "animatingToHeightAuto": "rah-animating--to-height-auto",
                "animatingToHeightSpecific": "rah-animating--to-height-specific",
                "animatingToHeightZero": "rah-animating--to-height-zero",
                "animatingUp": "rah-animating--up",
                "static": "rah-static",
                "staticHeightAuto": "rah-static--height-auto",
                "staticHeightSpecific": "rah-static--height-specific",
                "staticHeightZero": "rah-static--height-zero",
              }
            }
            applyInlineTransitions={true}
            delay={0}
            duration={500}
            easing="ease"
            height="auto"
            style={Object {}}
          >
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "paddingRight": 20,
                    "width": "60%",
                  }
                }
              >
                <div>
                  <LmcHandoverResidentItem
                    data={
                      Object {
                        "logs": Array [
                          Object {
                            "createdBy": "TestCarer",
                            "description": "This is a log",
                            "id": "TestId",
                          },
                        ],
                        "resident": Object {
                          "id": "TestId2",
                          "name": "TestResident",
                          "picture": "TestPicture",
                        },
                      }
                    }
                  />
                </div>
              </div>
              <div
                style={
                  Object {
                    "flex": "1",
                    "width": "40%",
                  }
                }
              >
                <LmcHandoverNotes
                  notes={
                    Array [
                      Object {
                        "createdBy": "TestCarer",
                        "id": "TestId",
                        "note": "This is a note",
                      },
                    ]
                  }
                />
              </div>
            </div>
          </AnimateHeight>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LmcCurrentHandover
    logsByResident={
      Array [
        Object {
          "logs": Array [
            Object {
              "createdBy": "TestCarer",
              "description": "This is a log",
              "id": "TestId",
            },
          ],
          "resident": Object {
            "id": "TestId2",
            "name": "TestResident",
            "picture": "TestPicture",
          },
        },
      ]
    }
    notes={
      Array [
        Object {
          "createdBy": "TestCarer",
          "id": "TestId",
          "note": "This is a note",
        },
      ]
    }
  />,
}
`;
